seacrowd/sea_datasets/parallel_asian_treebank/parallel_asian_treebank.py

```diff
@@ -0,0 +1,167 @@
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
+        if self.config.data_dir is None:
+            raise ValueError(
+                "This is a local dataset. Please pass the data_dir kwarg to load_dataset."
+            )
+        else:
+            data_dir = self.config.data_dir
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "data_dir": data_dir,
+                    "split": "train"
+                },
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST,
+                gen_kwargs={
+                    "data_dir": data_dir,
+                    "split": "test"
+                },
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.VALIDATION,
+                gen_kwargs={
+                    "data_dir": data_dir,
+                    "split": "dev"
+                },
+            )
+        ]
+
+    def _generate_examples(self, data_dir, split: str):
+
+        if self.config.schema not in ["source", "seacrowd_t2t"]:
+            raise ValueError(f"Invalid config: {self.config.name}")
+
+        mapping_data = {}
+
+        for language in _LANGUAGES:
+            datas = open(f"{data_dir}/data_{_LANGUAGES_TO_FILENAME_LANGUAGE_CODE[language]}.txt.{split}", "r").readlines()
+
+            for line in datas:
+                id, sentence = line.split("\t")
+                sentence = sentence.rsplit()
+
+                if id not in mapping_data:
+                    mapping_data[id] = {}
+
+                mapping_data[id][language] = sentence
+        
+        combination_languages = list(combinations(_LANGUAGES, 2))
+
+        i = 0
+
+        for id in mapping_data:
+            for each_pair in combination_languages:
+                if each_pair[0] in mapping_data[id] and each_pair[1] in mapping_data[id]:
+                    yield i, {
+                        "id": f"{id}-{each_pair[0]}-{each_pair[1]}",
+                        "text_1": mapping_data[id][each_pair[0]],
+                        "text_2": mapping_data[id][each_pair[1]],
+                        "text_1_name": each_pair[0],
+                        "text_2_name": each_pair[1],
+                    }
+
+                    i+=1
+
+                    yield i, {
+                        "id": f"{id}-{each_pair[1]}-{each_pair[0]}",
+                        "text_1": mapping_data[id][each_pair[1]],
+                        "text_2": mapping_data[id][each_pair[0]],
+                        "text_1_name": each_pair[1],
+                        "text_2_name": each_pair[0],
+                    }
+
+                    i+=1
```

Comment:
I'm not sure if yielding a reverse pair is necessary 🤔 Are they the same but with the keys switched? If yes, I think yielding one combination should suffice (I may be wrong though so feel free to ignore)